### PR TITLE
Explicitly configure HTTP request for PayPal

### LINKED ISSUE #8878: gift subscriptions and gift gems can't be purchased through PayPal
Trying to send a gift subscription results in a "Server currently unreachable" notification as soon as you click on the PayPal button. The PayPal website is available at the same time.  Creating a non-gift subscription for yourself does not produce this error, nor does attempting to buy gems with PayPal.

Thanks to XBZ1337 (ed9bd06a-742b-4a6c-be3e-f06f49a564ba) for reporting this in the Report a Bug guild.

### PR DESCRIPTION
**Pull request by SabreCat**
[//]: # (Note: See http://habitica.wikia.com/wiki/Using_Habitica_Git#Pull_Request for more info)

[//]: # (Put Issue # or URL here, if applicable. This will automatically close the issue if your PR is merged in)
Fixes #8878

### Changes
[//]: # (Describe the changes that were made in detail here. Include pictures if necessary)

Implements @Audiotuete's suggestion from https://github.com/HabitRPG/habitica/issues/8878#issuecomment-317247133, explicitly setting headers for the HTTP GET call to PayPal when sending a transaction.

Though this is a client-side fix, which would otherwise be on hold due to our frontend redesign underway, it addresses an issue blocking a revenue stream and is thus worthy of a hotfix.

[//]: # (Put User ID in here - found in Settings -> API)

### CODE REVIEW

File: website/client-old/js/services/paymentServices.js [hunk 1]
```diff
@@ -362,7 +362,15 @@ function($rootScope, User, $http, Content, Notification) {
...
-    $http.get(url);
+    var req = {
+      method: 'GET',
+      url: url,
+      headers: {
+        'Content-Type': 'text/plain;charset=utf-8',
+      },
+    };
+
+    $http(req);
```
Comment (TheHollidayInn) [inline website/client-old/js/services/paymentServices.js]: Let's return this just in case.

**Review by TheHollidayInn — CHANGES_REQUESTED** [review]

**Review by TheHollidayInn — APPROVED** [review]

**Comment by SabreCat** [comment]
Implemented @Audiotuete's follow-up suggestion https://github.com/HabitRPG/habitica/issues/8878#issuecomment-317598329 . @TheHollidayInn, mind rechecking this? I've done a functional test and it does appear to address the issue.

**Review by TheHollidayInn — APPROVED** [review]

**Comment by SabreCat** [comment]
@Audiotuete, what's your User ID in Habitica? I'd like to give you contributor credit for this, as you did the actual issue investigation, I only did the logistics of getting it into the code.

**Comment by Audiotuete** [comment]
Thanks, my ID is:

ca8a6f62-d996-4f8a-a0eb-37bd7a3bc3a6

**Comment by SabreCat** [comment]
@Audiotuete Blacksmith tier awarded! To get to Tier 2 you'll likely need to get your own PRs up and running, but we do appreciate the help!

**Pull request merged**